test/ts/unit/simple_interest_terms_contract.ts

```diff
@@ -0,0 +1,445 @@
+        termsContract = new SimpleInterestTermsContractContract(termsContractWeb3Contract, TX_DEFAULTS);
+
+        // PAYER begins with a balance of 5 ether.
+        await mockToken.mockBalanceOfFor.sendTransactionAsync(
+            PAYER, Units.ether(5),
+        );
+
+        // TransferProxy is granted an allowance of 3 ether from PAYER.
+        await mockToken.mockAllowanceFor.sendTransactionAsync(
+            PAYER, mockTokenTransferProxy.address, Units.ether(3),
+        );
+
+    });
+
+    describe("Initialization", () => {
+        it("points to the DebtRegistry passed in through the constructor", async () => {
+            await expect(termsContract.debtRegistry.callAsync()).to.eventually.equal(mockRegistry.address);
+        });
+        it("points to the RepaymentRouter passed in through the constructor", async () => {
+          await expect(termsContract.repaymentRouter.callAsync()).to.eventually.equal(router.address);
+        });
+        it("points to the token contract passed in through the constructor", async () => {
+          await expect(termsContract.repaymentToken.callAsync()).to.eventually.equal(mockToken.address);
+        });
+    });
+
+    describe("#registerRepayment", () => {
+
+        describe("agent who is not RepaymentRouter calls registerRepayment", () => {
+
+            it("should throw", async () => {
+              await expect(termsContract.registerRepayment.sendTransactionAsync(
+                ARBITRARY_AGREEMENT_ID,
+                PAYER,
+                BENEFICIARY,
+                Units.ether(1),
+                mockToken.address,
+                { from: ATTACKER } // the message is coming from an attacker.
+              )).to.eventually.be.rejectedWith(REVERT_ERROR);
+            });
+
+        });
+
+        describe("RepaymentRouter calls registerRepayment", () => {
+
+            before(async () => {
+                await mockRegistry.mockGetBeneficiaryReturnValueFor.sendTransactionAsync(
+                    ARBITRARY_AGREEMENT_ID,
+                    BENEFICIARY,
+                );
+
+                await mockRegistry.mockGetTermsReturnValueFor.sendTransactionAsync(
+                    ARBITRARY_AGREEMENT_ID,
+                    termsContract.address,
+                    TERMS_CONTRACT_PARAMETERS,
+                );
+
+                await mockRegistry.mockGetTermsContractReturnValueFor.sendTransactionAsync(
+                    ARBITRARY_AGREEMENT_ID,
+                    termsContract.address,
+                );
+            });
+
+            describe("...with a different `tokenAddress` than expected by the terms contract", () => {
+
+                before(async () => {
+                    await expect(router.repay.sendTransactionAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        Units.ether(1),
+                        ATTACKER // this is not the address it's expecting.
```

Comment:
Though this test should pass correctly, the `REVERT_ERROR` might be thrown for different reasons than what we're trying to test for.  The account at `ATTACKER` has neither a contract deployed to it, nor an ERC20 compliant one at that, so there are a bunch of reasons _other_ than `ATTACKER != tokenAddress` that this would throw.